### PR TITLE
Misc fixes and cleanups to Debian packaging and Debian packaging testing

### PR DESCRIPTION
**Pull request by ottok**

### CODE REVIEW

**Comment by ottok** [comment]
@cvicentiu I found minor regressions, marked this as WIP until I have force pushed the final version. Feel free to comment contents anyway.

**Comment by ottok** [comment]
This can be reviewed and merged now.

The Gitlab-CI shows currently this:

image

It will be all green when https://github.com/MariaDB/server/pull/1496 is final (and also https://github.com/MariaDB/server/pull/1478 and https://github.com/MariaDB/server/pull/1453 are merged).

## Tests

buildbot/opensuse-42 failed after mtr with:
```
process killed by signal 13
program finished with exit code -1
elapsedTime=388.673035
```
Most unlikely related to changes in this PR.

buildbot/pc9-clang10-ubuntu-1804 fails with same errors as explained in this PR on the 10.4 branch https://github.com/MariaDB/server/pull/1453#issuecomment-612342382, so they are unrelated to this PR.

**Comment by ottok** [comment]
Package contents comparison of this branch vs. 10.5 head:
```
$ diff -u filelist-2f7d91bb6ce-10.5.log filelist-eb52444133c-ok-10.5.log
--- filelist-2f7d91bb6ce-10.5.log	2020-04-14 21:34:13.164895355 +0300
+++ filelist-eb52444133c-ok-10.5.log	2020-04-14 22:08:40.604216767 +0300
@@ -605,6 +605,7 @@
 -rwxr-xr-x root/root ./usr/bin/mariadb-import
 lrwxrwxrwx root/root ./usr/bin/mariadb-optimize
 lrwxrwxrwx root/root ./usr/bin/mariadb-repair
+-rwxr-xr-x root/root ./usr/bin/mariadb-report
 -rwxr-xr-x root/root ./usr/bin/mariadb-show
 -rwxr-xr-x root/root ./usr/bin/mariadb-slap
 -rwxr-xr-x root/root ./usr/bin/mariadb-waitpid
@@ -645,6 +646,7 @@
 lrwxrwxrwx root/root ./usr/share/man/man1/mariadb-import.1.gz
 lrwxrwxrwx root/root ./usr/share/man/man1/mariadb-optimize.1.gz
 lrwxrwxrwx root/root ./usr/share/man/man1/mariadb-repair.1.gz
+-rw-r--r-- root/root ./usr/share/man/man1/mariadb-report.1.gz
 lrwxrwxrwx root/root ./usr/share/man/man1/mariadb-show.1.gz
 lrwxrwxrwx root/root ./usr/share/man/man1/mariadb-slap.1.gz
 lrwxrwxrwx root/root ./usr/share/man/man1/mariadb-waitpid.1.gz
@@ -658,7 +660,7 @@
 -rw-r--r-- root/root ./usr/share/man/man1/mysqlimport.1.gz
 lrwxrwxrwx root/root ./usr/share/man/man1/mysqloptimize.1.gz
 lrwxrwxrwx root/root ./usr/share/man/man1/mysqlrepair.1.gz
--rw-r--r-- root/root ./usr/share/man/man1/mysqlreport.1.gz
+lrwxrwxrwx root/root ./usr/share/man/man1/mysqlreport.1.gz
 -rw-r--r-- root/root ./usr/share/man/man1/mysqlshow.1.gz
 -rw-r--r-- root/root ./usr/share/man/man1/mysqlslap.1.gz
 -rw-r--r-- root/root ./usr/share/man/man1/mysql_waitpid.1.gz
@@ -923,6 +925,7 @@
 -rwxr-xr-x root/root ./usr/bin/wsrep_sst_mariabackup
 -rwxr-xr-x root/root ./usr/bin/wsrep_sst_mysqldump
 -rwxr-xr-x root/root ./usr/bin/wsrep_sst_rsync
+lrwxrwxrwx root/root ./usr/bin/wsrep_sst_rsync_wan
 drwxr-xr-x root/root ./usr/lib/
 drwxr-xr-x root/root ./usr/lib/mysql/
 drwxr-xr-x root/root ./usr/lib/mysql/plugin/
@@ -1000,6 +1003,7 @@
 -rw-r--r-- root/root ./usr/share/man/man1/wsrep_sst_mariabackup.1.gz
 -rw-r--r-- root/root ./usr/share/man/man1/wsrep_sst_mysqldump.1.gz
 -rw-r--r-- root/root ./usr/share/man/man1/wsrep_sst_rsync.1.gz
+-rw-r--r-- root/root ./usr/share/man/man1/wsrep_sst_rsync_wan.1.gz
 drwxr-xr-x root/root ./usr/share/mysql/
 -rwxr-xr-x root/root ./usr/share/mysql/debian-start.inc.sh
 -rwxr-xr-x root/root ./usr/share/mysql/echo_stderr
@@ -1154,6 +1158,7 @@
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/debug_key_management.so
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/dialog_examples.so
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/example_key_management.so
+-rw-r--r-- root/root ./usr/lib/mysql/plugin/func_test.so
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/ha_example.so
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/ha_test_sql_discovery.so
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/libdaemon_example.so
@@ -1161,6 +1166,7 @@
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/qa_auth_interface.so
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/qa_auth_server.so
 -rw-r--r-- root/root ./usr/lib/mysql/plugin/test_versioning.so
+-rw-r--r-- root/root ./usr/lib/mysql/plugin/type_test.so
 drwxr-xr-x root/root ./usr/lib/x86_64-linux-gnu/
 drwxr-xr-x root/root ./usr/lib/x86_64-linux-gnu/libmariadb3/
 drwxr-xr-x root/root ./usr/lib/x86_64-linux-gnu/libmariadb3/plugin/
```

**Comment by ottok** [comment]
Rebased on latest 10.5 upstream. In addition to the CI reports visible on Github there are also:
- http://buildbot.askmonty.org/buildbot/grid?branch=ok-10.5&category=main&category=experimental
- https://salsa.debian.org/mariadb-team/mariadb-server/pipelines/127266

File: debian/additions/mariadb-report [hunk 1]
```diff
@@ -956,14 +957,14 @@ sub make_short
 #
 # The format_u_time sub simply beautifies long_query_time.
 
-sub format_u_time  # format microsecond (�) time value
+sub format_u_time  # format microsecond (�) time value
 {
```
Comment (vuvova) [inline debian/additions/mariadb-report:961]: you seem to have corrupted `µ` in the comments (here and below)

File: debian/autobake-deb.sh [hunk 2]
```diff
@@ -112,34 +105,40 @@ LOGSTRING="MariaDB build"
...
-# On Travis CI, use -b to build binary only packages as there is no need to
-# waste time on generating the source package.
+# On Travis CI and Gitlab-CI, use -b to build binary only packages as there is
+# no need to waste time on generating the source package.
 if [[ $TRAVIS ]]
 then
   BUILDPACKAGE_FLAGS="-b"
 fi
 
+# Use eatmydata is available to build faster with less I/O, skipping fsync()
```
Comment (vuvova) [inline debian/autobake-deb.sh:119]: who's fsyncing?

Comment (ottok) [reply debian/autobake-deb.sh:119]: Any step in the entire build. Eatmydata is a generic performance optimizer for use cased where errors are not critical, such when running a development build (if my laptop looses power during the build it does not matter, no critical data).

File: debian/additions/mariadb-report (hunk 1 above)
Comment (ottok) [reply debian/additions/mariadb-report:961]: Thanks for noticing! Fixed all 3 occurrences.

File: debian/autobake-deb.sh (hunk 2 above)
Comment (ottok) [reply debian/autobake-deb.sh:119]: Updated inline comment to be more clear about this

**Comment by cvicentiu** [comment]
@ottok I don't have any particular comments. Changes look good.

I'll wait for you to finish the WIP (looks done to me), I will run a set of tests just to be sure, then I will merge.

Now, the Gitlab CI is really cool. Unfortunately, if we don't integrate it with developer workflows it will be broken and ignored. I will investigate with @vladbogo if we can integrate a view of that within buildbot.mariadb.org, or if we can reproduce it as part of our buildbot pipeline.

In the meantime, keeping the files in won't actively hurt.

**Comment by ottok** [comment]
Hello!

We can have a Gitlab mirror of gitlab.com/mariadb/server that automatically
syncs the protected branches (10.5, 10.4 etc) and runs Gitlab-CI on them.
Each developer will get feedback via email based on their author/commit
email.

Naturally we need to ask developers first if they want to participate in
such an experiment.

**Comment by ottok** [comment]
@cvicentiu OK, this can be merged now.

**Comment by ottok** [comment]
Checked buildbot.mariadb.org statuses above, timeouts and mtr failures not related to this PR.

**Pull request merged**